src/PrestaShopBundle/Resources/config/services/adapter/product.yml

```diff
@@ -69,6 +69,20 @@ services:
             - name: tactician.handler
               command: PrestaShop\PrestaShop\Core\Domain\Product\Query\GetProductIsEnabled
 
+    prestashop.adapter.product.image_thumbnail_provider:
+        class: 'PrestaShop\PrestaShop\Adapter\Product\ImageThumbnailProvider'
+        arguments:
+            - '@prestashop.core.image.parser.image_tag_source_parser'
+
+    prestashop.adapter.product.grid.data.factory.product_price_grid_data_factory_decorator:
+        class: 'PrestaShop\PrestaShop\Adapter\Product\Grid\Data\Factory\ProductPriceGridDataFactoryDecorator'
+        arguments:
+            - '@prestashop.core.grid.data.factory.product'
+            - '@prestashop.core.localization.locale.repository'
+            - '@=service("prestashop.adapter.legacy.context").getContext().language.getLocale()'
+            - '@=service("prestashop.adapter.legacy.configuration").get("PS_CURRENCY_DEFAULT")'
```

Comment:
So even if BO user is using French language for BO, if default currency is US dollar, he will see products in US dollar, right?

I guess this is how it's done in current page?

Reply:
Yes.

IMO it wouldn't make sense to tie currency choice to language or even country. It should always be default currency unless we introduce option for employees to change their currency.